kernel/mem.c

```diff
@@ -55,23 +55,28 @@ void mem_init(void)
         mem_size);
 }
 
-/*
- * Called by dlmalloc to allocate or free memory.
+/* 
+ * Allocate a 4K aligned chunk on the brk.  Should only be called
+ * before solo5_app_main. 
  */
-void *sbrk(intptr_t increment)
+void *alloc_chunk_4K(size_t size)
 {
     uint64_t prev, brk;
     uint64_t heap_max = (uint64_t)&prev - stack_guard_size;
     prev = brk = heap_top;
 
-    /*
-     * dlmalloc guarantees increment values less than half of size_t, so this
-     * is safe from overflow.
-     */
-    brk += increment;
+    /* for 4K alignment */
+    prev = brk = brk + (0x1000 - (brk & 0xfff));
+
+    brk += size;
```

Comment:
This can overflow. There should not be a `size` greater than 1k or something like that right? you could assert that.

Reply:
The name suggests that the memory returned will be rounded up to 4k. No caller is expecting that right?